### PR TITLE
refactor: remove signing and signature verification of data posted to DA

### LINKED ISSUE #523: remove signing of celestia data posting
Currently we sign the data which is posted to Celestia and then verify that signature, in addition to verifying the sequencer block data.

We don't need to care who posted the data to celestia, only that it is the correct Sequencer data.

### PR DESCRIPTION
**Pull request by noot**
## Summary
remove signing and signature verification of data posted to DA so that data can be relayed to DA by anyone, not just the block proposer.

## Background
We want validators to not have to also run a relayer. The sequencer block data is already committed by the sequencer chain, so additional signatures aren't necessary. Also, you can already see who signed+submitted the data by reading the signature of the Celestia tx that submitted it.

## Changes
- remove `SignedNamespaceData` and all mentions/usages
- update relayer to have a `disable_relay_all` option, which, when set to true, relays only blocks signed by the given validator key. the default value is `false`, so all blocks are relayed.
- update conductor to verify the block received from DA has the same hash as the canonical block at that height returned by the sequencer

## Testing
unit tests

## Breaking Changelist
- data posted to DA is no longer in the form of `SignedNamespaceData`, so deserializing previously-written data will fail.

## Related Issues

closes #523

### CODE REVIEW

**Review by sambukowski — APPROVED** [review]

**Review by SuperFluffy — CHANGES_REQUESTED** [review]
I believe with this change we are no longer guarded against a malicious sequencer-relayer submitting `SequencerNamespaceData` blobs with omitted chain IDs (and a new chain ID root) to celestia.

Before the assumption was:
1. sequencer-relayer runs as sidecar to cometbft/sequencer
2. cometbft/sequencer are fully trusted
3. hence all blobs submitted to celestia by sequencer-relayer are fully trusted (if signature + public key check out)

Now that the signing part is gone the blobs can no longer automatically trusted.

However, from what I can see the chain ID commitment is not included in the construction of the overall data hash, nor in its verification. It's only verified that the chain ID commitment is correct given the list of chain IDs (which does not give sufficient protection).

Hence I think this PR should do one of the following:
1. include the chain ID commitment in the construction/verification of the overall `data_hash`
2. or alternatively verify that the blob was submitted to celestia by the expected party to ensure that the list of chain ID can be trusted.

**Comment by noot** [comment]
@SuperFluffy see https://github.com/astriaorg/astria/pull/548 which addresses the issue

**Review by joroshiba — APPROVED** [review]

**Review by sambukowski — APPROVED** [review]

**Comment by noot** [comment]
@SuperFluffy #548 was merged which addresses your concerns, feel free to re-review this!

File: crates/astria-celestia-client/src/client.rs [hunk 1]
```diff
@@ -288,28 +283,22 @@ fn assemble_blobs_from_sequencer_block_data(
     Ok(blobs)
 }
 
-/// Filters out blobs that cannot be deserialized to `SignedNamespaceData<RollupNamespaceData>`,
-/// whose block hash or public key do not match that of `sequencer_data`, respectively, or that
+/// Filters out blobs that cannot be deserialized to `RollupNamespaceData`,
+/// whose block hash does not match that of `sequencer_data`, respectively, or that
```
Comment (SuperFluffy) [inline crates/astria-celestia-client/src/client.rs:287]:
```suggestion
/// whose block hash does not match that of `sequencer_data`, or that
```

File: crates/astria-composer/tests/blackbox/composer.rs [hunk 2]
```diff
@@ -5,7 +5,9 @@ use proto::generated::sequencer::v1alpha1::NonceResponse;
 use sequencer_types::AbciCode;
 use tendermint_rpc::{
     endpoint::broadcast::tx_sync,
-    request::{self,},
+    request::{
```
Comment (SuperFluffy) [inline crates/astria-composer/tests/blackbox/composer.rs:8]:
```suggestion
    request,
```
I think this should work? Importing `request::self` was probably an oversight of a refactor.

File: crates/astria-composer/tests/blackbox/composer.rs [hunk 3]
```diff
@@ -5,7 +5,9 @@ use proto::generated::sequencer::v1alpha1::NonceResponse;
...
+        self,
```
Comment (SuperFluffy) [inline crates/astria-composer/tests/blackbox/composer.rs:9]: please fix the import, see comment above

File: crates/astria-proto/proto/astria/sequencer/v1alpha1/data.proto
Comment (SuperFluffy) [inline crates/astria-proto/proto/astria/sequencer/v1alpha1/data.proto:1]: Was this even used?

File: crates/astria-sequencer-relayer/local.env.example
Comment (SuperFluffy) [inline crates/astria-sequencer-relayer/local.env.example:1]: This change seems to be orthogonal to the refactor itself?

File: crates/astria-sequencer-relayer/src/relayer.rs [hunk 4]
```diff
@@ -41,9 +41,8 @@ pub(crate) struct Relayer {
     // The limit that relayer will pay for submitting sequencer blocks to the DA.
     gas_limit: u64,
 
-    // Carries the signing key to sign sequencer blocks before they are submitted to the data
-    // availability layer.
-    validator: Validator,
+    // If this is set, only relay blocks to DA which are proposed by the same validator key.
+    validator: Option<Validator>,
```
Comment (SuperFluffy) [inline crates/astria-sequencer-relayer/src/relayer.rs:45]: How does this relate to the refactor?

**Review by SuperFluffy — CHANGES_REQUESTED** [review]
I was about to give approval when I encountered `ASTRIA_SEQUENCER_RELAYER_DISABLE_RELAY_ALL`.  Can you clarify in how far this is related to the refactor, i.e. to the removal of the `SignedNamespaceData` wrapper?

If it is not related, please move this to a separate PR?

**Comment by noot** [comment]
@SuperFluffy the `ASTRIA_SEQUENCER_RELAYER_DISABLE_RELAY_ALL` exists to optionally keep the functionality where the relayer only relays block proposed by a specific proposer (as before). @joroshiba wanted to keep this ?

File: crates/astria-sequencer-relayer/local.env.example
Comment (joroshiba) [reply crates/astria-sequencer-relayer/local.env.example:1]: I don't think it is from a product standpoint. The product level ask is to make it so every validator doesn't need to run a relayer, and enable running a single validator.

The removal of signature is the technical detail of what is required for that. I think if anything the ticket could use some better description.

**Review by joroshiba — APPROVED** [review]
some lint issues

File: crates/astria-proto/proto/astria/sequencer/v1alpha1/data.proto
Comment (noot) [reply crates/astria-proto/proto/astria/sequencer/v1alpha1/data.proto:1]: don't think so, should i just remove the whole file?

File: crates/astria-sequencer-relayer/src/relayer.rs (hunk 4 above)
Comment (noot) [reply crates/astria-sequencer-relayer/src/relayer.rs:45]: see my reply - it's to optionally keep the functionality as previously, where we only relay blocks from a certain proposer

**Comment by noot** [comment]
@joroshiba proto lint is going to fail because the proto files were changed, other fmt issue should be fixed

File: crates/astria-proto/proto/astria/sequencer/v1alpha1/data.proto
Comment (SuperFluffy) [reply crates/astria-proto/proto/astria/sequencer/v1alpha1/data.proto:1]: I think you can just remove the diff - i.e. don't touch the protobuf at all because it does not matter. This way we won't have a breaking change with buf and can just leave the breaking changes to #444 (once it lands.... soon).

File: crates/astria-sequencer-relayer/local.env.example
Comment (SuperFluffy) [reply crates/astria-sequencer-relayer/local.env.example:1]: Yeah, thanks. I got it now

File: crates/astria-sequencer-relayer/src/relayer.rs [hunk 5]
```diff
@@ -93,8 +92,20 @@ impl Relayer {
         let sequencer = HttpClient::new(&*cfg.sequencer_endpoint)
             .wrap_err("failed to create sequencer client")?;
 
-        let validator = Validator::from_path(&cfg.validator_key_file)
-            .wrap_err("failed to get validator info from file")?;
+        let validator = if cfg.disable_relay_all {
```
Comment (SuperFluffy) [inline crates/astria-sequencer-relayer/src/relayer.rs:95]: This can be expressed a bit neater without right-drift if-let chains.
```rust
let validator = match (cfg.disable_relay_all, cfg.validator_key_file) {
    (true, Some(file)) => Some(validator::from_path(file).wrap_err("baaaaad")?),
    (true, None) => bail!("disable relay set, but no file, etc etc"),
    (false, _) => None, // could also say that the file was unnecessarily set, but it's ok
};

File: crates/astria-sequencer-relayer/local.env.example [hunk 6]
```diff
@@ -19,8 +19,12 @@ ASTRIA_SEQUENCER_RELAYER_GAS_LIMIT=1000000
 # requesting a new block from sequencer.
 ASTRIA_SEQUENCER_RELAYER_BLOCK_TIME=1000
 
+# Set to true to only relay blocks proposed by the validator key in `ASTRIA_SEQUENCER_RELAYER_VALIDATOR_KEY_FILE`.
+ASTRIA_SEQUENCER_RELAYER_DISABLE_RELAY_ALL=false
+
 # The path to the file storing the cometbft validator key.
 # Used to sign data before submitting it to celestia.
+# Ignored if `ASTRIA_SEQUENCER_RELAYER_DISABLE_RELAY_ALL=false`.
```
Comment (SuperFluffy) [inline crates/astria-sequencer-relayer/local.env.example:27]: When I made the suggestion to express validator construction in another review comment I realized that this naming confused me.

Can we switch the logic and call this `ASTRIA_SEQUENCER_RELAYER_RELAY_ONLY_BLOCKS_MATCHING_VALIDATOR_KEY`? or `*_RELAY_ONLY_VALIDATOR_KEY_BLOCKS`?

I know it's wordy, but easier to grok.

**Review by SuperFluffy — APPROVED** [review]
The changes look good. Just some minor comments.

I would like to see two blackbox tests for the config option. Right now in the mock we are mounting responses that are signed with the key.

For tests, we could send a block without a wrong signature: send two blocks, 1st with a a bad, 2nd with a good signature. For relaying all disabled => we only observe the 2nd block. With relaying all enabled => we observe both blocks.

File: crates/astria-sequencer-relayer/src/relayer.rs (hunk 5 above)
Comment (noot) [reply crates/astria-sequencer-relayer/src/relayer.rs:95]: oh cool :D
